### DISCUSSION TITLE
Join 2 json data points using a common "field"

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana-infinity-datasource#542] [Ideas]
Greetings,

This feature request is to be able to join, like you would in sql query, two distinct json data points using a common field they share.

For example we have the following geojson file https://d2ad6b4ur7yvpq.cloudfront.net/naturalearth-3.3.0/ne_50m_rivers_lake_centerlines_scale_rank.geojson. It lists out rivers and lakes around the world. 

```
{
	"type": "Feature",
	"properties": {
		"strokeweig": 0.200000003,
		"scalerank": 5,
		"name": "Mississippi",
		"dissolve": "Lake CenterlineMississippi",
		"note": null,
		"featureclass": "Lake Centerline"
	},
	"geometry": {
		"type": "LineString",
		"coordinates": [
			[
				-94.50680986202217,
				47.45640595152052
			],
			[
				-94.63706030961385,
				47.43999868426785
			]
		]
	}
}
```
But notice it is missing depth, width or square meter and country name. So we have another json data that has these values as follows

```
{
	"country": "USA",
	"name": "Mississippi",
	"depth": 12,
	"square_meter": 7
}
```

The common "field" would be name. So by joining on that "field" we have enriched the data to include something it was originally missing but might be provided by another data point/api.

Thanks

**Comment by yesoreyeram** [comment]
Thanks @yosiasz for writing this. Seems to be good use case for joins. I couldn't think anything on top of my mind. But added to my [backlog](https://github.com/yesoreyeram/grafana-infinity-datasource/issues/530) of items to work on.
